modules/extensions-spring-support/src/test/resources/petstore-expression-config-ref.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<mule xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      xmlns:petstore="http://www.mulesoft.org/schema/mule/petstore"
+      xmlns="http://www.mulesoft.org/schema/mule/core"
+      xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd
+               http://www.mulesoft.org/schema/mule/petstore http://www.mulesoft.org/schema/mule/petstore/current/mule-petstore.xsd">
+
+    <petstore:config name="john-store" cashierName="jim">
+        <petstore:connection username="john" password="doe"/>
+        <petstore:pets>
+            <petstore:pet value="Lassie"/>
+            <petstore:pet value="Sapo Pepe"/>
+            <petstore:pet value="My mother's parrot"/>
+        </petstore:pets>
+    </petstore:config>
+
+    <petstore:config name="paw-patrol-store" cashierName="rocky">
+        <petstore:connection username="john" password="doe"/>
+        <petstore:pets>
+            <petstore:pet value="Marshall"/>
+            <petstore:pet value="Chase"/>
+            <petstore:pet value="Skye"/>
+        </petstore:pets>
+    </petstore:config>
+
+    <flow name="getPetsWithReference">
+        <petstore:get-pets ownerName="john" config-ref="paw-patrol-store"/>
+    </flow>
+
+    <flow name="getPetsWithExpression">
+        <petstore:get-pets ownerName="john" config-ref="#[app.registry[vars.storeName ++ '-store']]"/>
```

Comment:
I think it should not be necessary to access the registry, right? Probably the result of the expression should be automatically used as a bean reference. What do you think?